Enhance search bar UX

danielcolinjames opened the pull request:
- addresses #735
- add loading state
- add "No results" state
- only allow pressing Enter after loading results is complete

danielcolinjames committed:
Enhance search bar UX
- addresses #735
- add loading state
- add "No results" state
- only allow pressing Enter after loading results is complete

cokes518 commented:
expand the search box, add padding to the address so it doesn't appear behind the icon.

<img width="468" alt="image" src="https://user-images.githubusercontent.com/19331274/140104380-89a76418-0664-4a38-b874-db3733d9de3c.png">

danielcolinjames committed:
More search enhancements
- add dark blur layer behind results on mobile
- expand search bar when focused

danielcolinjames commented:
added some nice improvements for mobile:
![2021-11-03 17 45 11](https://user-images.githubusercontent.com/10648471/140238672-6a9e54e0-f8a4-475c-8427-cd69ebdef6ef.gif)

and for desktop:
![2021-11-03 17 47 26](https://user-images.githubusercontent.com/10648471/140238844-3bd11a12-62a4-495d-90d1-49ea1f1cd009.gif)

cokes518 commented:
🎈
________________________________
From: Daniel James ***@***.***>
Sent: Wednesday, November 3, 2021 5:48:42 PM
To: helium/explorer ***@***.***>
Cc: Coco Tang ***@***.***>; Comment ***@***.***>
Subject: Re: [helium/explorer] Enhance search bar UX (PR #739)


added some nice improvements for mobile:
[2021-11-03 17 45 11]<https://user-images.githubusercontent.com/10648471/140238672-6a9e54e0-f8a4-475c-8427-cd69ebdef6ef.gif>

and for desktop:
[2021-11-03 17 47 26]<https://user-images.githubusercontent.com/10648471/140238844-3bd11a12-62a4-495d-90d1-49ea1f1cd009.gif>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/helium/explorer/pull/739#issuecomment-960345295>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AETPRSQ3MUUCAH64ZJYAXELUKHQ7VANCNFSM5HHRHQCA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

danielcolinjames committed:
Improve UX / visuals

allenan commented on components/SearchBar/SearchBar.js:
```diff
@@ -102,45 +115,93 @@ const SearchBar = () => {
+                title="Loading results..."
+                subtitle="Talking to the API..."
+              />
+            ) : results.length === 0 ? (
```
nested ternaries are a bit of an antipattern. could we extract a component out of here where we could then use traditional logical operators?

danielcolinjames replied:
good call 👍

allenan commented on components/SearchBar/useSearchResults.js:
```diff
@@ -222,27 +238,29 @@ const useSearchResults = () => {
-        // searchCities(term)
-        searchMapAddresses(term)
-        searchMaker(term)
+        await searchHotspot(term.replace(/-/g, ' '))
```
this is going to perform each of these searches sequentially. is that what we want?

danielcolinjames replied:
good catch, wrapped in a Promise.all

allenan commented:
really nice ux enhancement for mobile 🚀

danielcolinjames committed:
Componentize results

danielcolinjames committed:
Merge branch 'v2' into search-bar-enhancements

danielcolinjames committed:
Add better padding for the cancel search button

danielcolinjames committed:
Do searches in parallel

allenan reviewed (approved):
revisions look good
